front/lib/api/csv.ts

```diff
@@ -1,5 +1,98 @@
 import { parse } from "csv-parse";
 
+const POSSIBLE_VALUES_MAX_LEN = 32;
+const POSSIBLE_VALUES_MAX_COUNT = 16;
+
+interface ColumnTypeInfo {
+  type: "string" | "number" | "boolean" | "enum";
+  possibleValues: string[] | null;
+}
+
+interface CSVRow {
+  [key: string]: string;
+}
+
+/**
+ * Detect the type of a column based on its values.
+ */
+function detectColumnType(columnValues: Iterable<string>): ColumnTypeInfo {
+  const uniqueValues = new Set<string>();
+  let allNumbers = true;
+  let allBooleans = true;
+
+  for (const value of columnValues) {
+    uniqueValues.add(value);
+
+    if (allNumbers && isNaN(Number(value))) {
+      allNumbers = false;
+    }
+
+    if (
+      allBooleans &&
+      !(value.toLowerCase() === "true" || value.toLowerCase() === "false")
+    ) {
+      allBooleans = false;
+    }
+
+    if (
+      !allNumbers &&
+      !allBooleans &&
+      uniqueValues.size > POSSIBLE_VALUES_MAX_COUNT
+    ) {
+      break; // Early exit if no further checks are needed
+    }
+  }
+
+  if (allNumbers) {
+    return { type: "number", possibleValues: null };
```

Comment:
IMO if uniqueValues.size <= POSSIBLE_VALUES_MAX_COUNT we should include

Reply:
Should be fixed!